int8[]  sphere_status